### PR TITLE
Bishop valid move

### PR DESCRIPTION
**Pull request by mmeitl**
Introduced the valid_move? method for the Bishop.
New tests are also introduced that validates this new method.

The tests verify that the Bishop is only able to move in diagonally unobstructed? or capturable? paths.

### CODE REVIEW

File: app/models/bishop.rb [hunk 1]
```diff
@@ -1,4 +1,23 @@
 class Bishop < Piece
+  def valid_move?(destination_row, destination_column)
+    if diagonal?(destination_row, destination_column)
+      return true if capturable?(destination_row, destination_column) || (unobstructed?(destination_row, destination_column) && unoccupied_space?(destination_row, destination_column))
+      return false
```
Comment (mscoutermarsh) [inline app/models/bishop.rb]: could remove this line. will still work the same

File: app/models/bishop.rb [hunk 2]
```diff
@@ -1,4 +1,23 @@
...
+    end
+    false
+  end
+
+  def unobstructed?(destination_row, destination_column)
+    return true unless obstructed?(destination_row, destination_column)
+    false
+  end
```
Comment (mscoutermarsh) [inline app/models/bishop.rb:11]: nice.

another way of writing this with a single line would be :)

```
!obstructed?(destination_row, destination_column)
```

**Comment by mscoutermarsh** [comment]
@mmeitl this looks great! left a couple comments. Feel free to merge it whenever you're ready :)

**Comment by mmeitl** [comment]
Modified some logic based on feedback.

**Pull request merged**